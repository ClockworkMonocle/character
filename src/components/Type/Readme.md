Representing all typeface in the application.

```js
<div>
  <Type>Plain copy</Type>
  <Type element="p" variant="body">Body copy</Type>
  <Type element="p" variant="subtext">subtext</Type>
  <Type element="h1" variant="heading-1">Heading 1</Type>
  <Type element="h1" variant="display-1">Display 1</Type>
  <Type element="h2" variant="display-2">Display 2</Type>
  <Type color="attack-color">Attack colored text</Type>
</div>
```
