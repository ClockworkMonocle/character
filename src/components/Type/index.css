
.type {
  display: block;
  margin: 0;
  padding: 0;
  font-size: 1em;
  font-weight: 400;
  line-height: 1.3;
}

.type__body {
  display: block;
  padding-bottom: 1em;
}

.type__subtext {
  font-size: 12px;
  color: var(--grey-55);
}

.type__heading-1 {
  display: block;
  font-size: 12px;
  text-transform: uppercase;
  font-weight: bold;
}

.type__heading-2 {
  display: block;
  padding-bottom: 1em;
  font-weight: bold;
}

.type__display-1 {
  display: block;
  font-size: 2em;
  padding-bottom: 1em;
}

.type__display-2 {
  display: block;
  font-weight: bold;
  font-size: 3em;
  padding-bottom: 1em;
}
