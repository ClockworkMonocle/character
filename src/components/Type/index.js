import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';

import './index.css';

export default function Type({
  element,
  variant,
  color,
  className,
  children,
  ...props
}) {
  const Disp = element;
  const css = cn('type', `type__${variant}`, `type__${color}`, className);
  return (<Disp
    className={css}
    {...props}
  >{children}</Disp>);
}

Type.propTypes = {
  element: PropTypes.oneOf([
    'span',
    'p',
    'h1',
    'h2',
    'h3',
    'h4',
    'h5',
    'h6',
  ]),
  variant: PropTypes.oneOf([
    'default',
    'body',
    'subtext',
    'heading-1',
    'heading-2',
    'display-1',
    'display-2',
  ]),
  color: PropTypes.oneOf([
    'white',
    'black',
    'dark',
    'grey-10',
    'grey-14',
    'grey-40',
    'grey-55',
    'grey-75',
    'grey-85',
    'grey-90',
    'grey-95',
    'attack-color',
    'spell-color',
    'str-color',
    'dex-color',
    'con-color',
    'wis-color',
    'int-color',
    'cha-color',
    'dark-attack-color',
    'dark-spell-color',
    'dark-str-color',
    'dark-dex-color',
    'dark-con-color',
    'dark-wis-color',
    'dark-int-color',
    'dark-cha-color',
    'light-attack-color',
    'light-spell-color',
    'light-str-color',
    'light-dex-color',
    'light-con-color',
    'light-wis-color',
    'light-int-color',
    'light-cha-color',
  ]),
};

Type.defaultProps = {
  element: 'span',
  variant: 'default',
  color: 'black',
}
