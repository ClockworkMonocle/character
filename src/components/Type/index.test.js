import React from 'react';
import { shallow } from 'enzyme';
import Type from './index';

test('<Type /> renders normal copy', () => {
  expect(shallow(<Type>normal</Type>)).toMatchSnapshot();
});

test('<Type /> renders subtext', () => {
  expect(shallow(<Type variant="subtext">subtext</Type>)).toMatchSnapshot();
});

test('<Type /> renders body', () => {
  expect(shallow(<Type variant="body">body</Type>)).toMatchSnapshot();
});

test('<Type /> renders heading-1', () => {
  expect(shallow(<Type variant="heading-1">heading-1</Type>)).toMatchSnapshot();
});

test('<Type /> renders heading-2', () => {
  expect(shallow(<Type variant="heading-2">heading-2</Type>)).toMatchSnapshot();
});

test('<Type /> renders display-1', () => {
  expect(shallow(<Type variant="display-1">display-1</Type>)).toMatchSnapshot();
});

test('<Type /> renders display-2', () => {
  expect(shallow(<Type variant="display-2">display-2</Type>)).toMatchSnapshot();
});
