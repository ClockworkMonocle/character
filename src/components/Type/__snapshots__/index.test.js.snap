// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Type /> renders body 1`] = `
<span
  className="type type__body type__black"
>
  body
</span>
`;

exports[`<Type /> renders display-1 1`] = `
<span
  className="type type__display-1 type__black"
>
  display-1
</span>
`;

exports[`<Type /> renders display-2 1`] = `
<span
  className="type type__display-2 type__black"
>
  display-2
</span>
`;

exports[`<Type /> renders heading-1 1`] = `
<span
  className="type type__heading-1 type__black"
>
  heading-1
</span>
`;

exports[`<Type /> renders heading-2 1`] = `
<span
  className="type type__heading-2 type__black"
>
  heading-2
</span>
`;

exports[`<Type /> renders normal copy 1`] = `
<span
  className="type type__default type__black"
>
  normal
</span>
`;

exports[`<Type /> renders subtext 1`] = `
<span
  className="type type__subtext type__black"
>
  subtext
</span>
`;
